### DISCUSSION TITLE
Mirth Connect Database is growing Rapidly

### DISCUSSION THREAD

**Discussion by sriram288** [nextgenhealthcare/connect#6184] [Q&A]
we are using the Mirth Connect version is 4.3.0
The database of Mirth is Connected to Cloud SQL [PostgreSQL] in GCP [Google Cloud Platform] Which have a version of 12.16
So my concern is from last month onwards the mirth-connect database in PostgreSQL is growing rapidly day by day.
And also parallely, the Archived_wal_log is increasing and it's almost the double the size of original data

we have enabled the Automated backups & Point-in-time recovery & also enabled the Autoscaling of storage. So it increases the storage whenever the size of Database increases.

So after had a discussion and we have set the Data pruning at the Global level & Channel level 

Here's the Settings for that
- we go with settings of 3-hour intervals and a block size of 1000
- we didn't touch the Archive Settings

At the time of pruning, we observed the PostgreSQL Database storage is growing.
And also the archived_wal_log storage is increased mainly[For this one, i have attached screenshot before & after pruning started]
Then after completing the Pruning the Data storage is constant

Attached the Screenshots

Here's my question:
- is the database storage is increasing during the Pruning was running. Is it expected?

Can someone please help on this

![Screenshot 2024-04-30 185033](https://github.com/nextgenhealthcare/connect/assets/99025813/14bae6c3-a343-4ae8-be27-2005d3f0a2d9)

![Screenshot 2024-04-30 185454](https://github.com/nextgenhealthcare/connect/assets/99025813/ae6044ad-fbce-468e-8337-f5707dbd6709)

![Screenshot 2024-04-30 190802](https://github.com/nextgenhealthcare/connect/assets/99025813/8e61274e-6c9e-445a-89bd-8a35d5ecaa56)

![Screenshot 2024-04-30 190919](https://github.com/nextgenhealthcare/connect/assets/99025813/bb5f76e6-b3f8-4dae-a370-6e8fb51c0ca9)

**Comment by rjkroll** [comment]
In the Postgres database itself, do you have Autovacuum enabled?  Autovacuum needs to run regularly to remove dead tuples (updated or deleted rows) and reclaim space. If it is not running effectively, the database will continue growing even with pruning.

**Comment by sriram288** [comment]
yes, the Autovacuum is enabled for Psql Database

![image](https://github.com/nextgenhealthcare/connect/assets/99025813/f6d531ef-a73a-4521-8586-04ef0cccc293)

And After setting the Pruning, the DB is growing constantly whenever the first pruning was run

**Reply by pacmano1** [reply]
Mirth is just doiing database deletees.   Ask your question over on a postgres forum.   As I mentioned in your mirth forums post, postgres probaby needs tuning.

**Reply by pacmano1** [reply]
Please do not post both places without referencing the other place.
